### PR TITLE
Add specification for query rule test API 

### PR DESCRIPTION
**Pull request by kderusso**
Adds a specification for the query rules test API added in https://github.com/elastic/elasticsearch/pull/114168

### CODE REVIEW

File: specification/query_rules/test/QueryRulesetTestRequest.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * under the License.
+ */
+import { RequestBase } from '@_types/Base'
+import { Id } from '@_types/common'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+
+/**
+ * Creates or updates a query ruleset.
+ * @rest_spec_name query_rules.put_ruleset
```
Comment (pquentin) [inline specification/query_rules/test/QueryRulesetTestRequest.ts:26]:
```suggestion
 * @rest_spec_name query_rules.test
```

File: specification/query_rules/test/QueryRulesetTestResponse.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { integer } from '@_types/Numeric'
+import { QueryRulesetMatchedRule } from './types'
```
Comment (pquentin) [inline specification/query_rules/test/QueryRulesetTestResponse.ts:21]: We recommend keeping classes that are used only once in the same file as the class that uses it - it's allowed as this is not Java!

**Review by pquentin — COMMENTED** [review]
Thank you for this!

File: specification/query_rules/test/QueryRulesetTestRequest.ts (hunk 1 above)
Comment (kderusso) [reply specification/query_rules/test/QueryRulesetTestRequest.ts:26]: You caught me in a copy paste 😅

**Comment by pquentin** [comment]
Can you confirm whether there's a Elasticsearch YAML test for API or not?

**Comment by l-trotta** [comment]
I have a question about `match_criteria`: in the documentation it says that:

> Match criteria should match the keys defined in the `criteria.metadata` field of the rule.

so can it only be something like `field_name: field_value`? since criteria.metadata was defined as `string` in the spec. If so then we could use a more specific type?

**Comment by kderusso** [comment]
> Can you please confirm whether there's a Elasticsearch YAML test for API or not?

@pquentin Yes, the yaml REST tests may be found [here](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/rules/70_query_rule_test.yml). 

> so can it only be something like field_name: field_value? since criteria.metadata was defined as string in the spec. If so then we could use a more specific type, like Dictionary<string,string>?

@l-trotta We can't use `Dictionary<string,string>` because the Java code supports `Object` such as numeric input.

**Review by l-trotta — APPROVED** [review]

**Pull request merged**